### ISSUE TITLE
Optional parameters should not trigger xUnit1041

### ISSUE THREAD

**Issue by rwb196884** [xunit/xunit#2846]
I've been using this for years
```
    public class ServerAdminIdentityResourceTests : TestBase
    {
        public ServerAdminIdentityResourceTests(WebApplicationFactory<Program> factory, bool passTroughSecurity = true) : base(factory, passTroughSecurity)
        {
        }

```
and
```    public abstract class TestBase : IClassFixture<WebApplicationFactory<Program>>
    {
        protected const string JsonMediaType = "application/json";
        private const string IntegrationTestScheme = "IntegrationTest";

        protected readonly HttpClient _client;
        protected ConfigurationDbContext _configurationDbContext;
        protected IdentityDbContext _identityDbContext;

        public TestBase(WebApplicationFactory<Program> factory, bool passTroughSecurity = true)
        {
            // Arrange
            _client = factory.WithWebHostBuilder(builder =>
            {
                if (passTroughSecurity)
                {
                    builder.ConfigureTestServices(services =>
                    {
                        var serviceProvider = services.BuildServiceProvider();
                        _configurationDbContext = serviceProvider.GetService<ConfigurationDbContext>();
                        _identityDbContext = serviceProvider.GetService<PrestwoodIdentityDbContext>();

                        services.AddAuthentication(options =>
                        {
                            options.DefaultAuthenticateScheme = IntegrationTestScheme;
                            options.DefaultChallengeScheme = IntegrationTestScheme;
                            options.DefaultScheme = IntegrationTestScheme;
                        })
                        .AddScheme<AuthenticationSchemeOptions, IntegrationTestAuthenticationHandler>(IntegrationTestScheme, options => { });
                    });
                }

            }).CreateClient(new WebApplicationFactoryClientOptions
            {
                AllowAutoRedirect = false
            });

            _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(IntegrationTestScheme);

            factory.IdentityConfigurationDatabase.EnsureDeleted();
        }


```
but all of a sudden you've fucked it up:
> xUnit1041 Fixture argument 'passTroughSecurity' does not have a fixture source (if it comes from a collection definition, ensure the definition is in the same assembly as the test)	

What the fuck?

**Comment by philippedurocher** [comment]
I'm experiencing the same problem with 2.6.3

```
[CollectionDefinition(nameof(DatabaseTestsCollectionFixture))]
public class DatabaseTestsCollectionFixture : ICollectionFixture<TestingFixture>
{ }
```

```
[Collection(nameof(DatabaseTestsCollectionFixture))]
public class TestBase : IDisposable
{
	public TestBase(TestingFixture fixture)
	{
		Fixture = fixture;
		Fixture.CreateCheckPoint();
	}

	public TestingFixture Fixture { get; }

	public void Dispose()
	{
		GC.SuppressFinalize(this);
		Fixture.Reset();
	}
}
```

Error
Fixture argument 'fixture' does not have a fixture source (if it comes from a collection definition, ensure the definition is in the same assembly as the test) (https://xunit.net/xunit.analyzers/rules/xUnit1041)

**Comment by bradwilson** [comment]
@philippedurocher Using your code as a template, I tried this, and it works as expected:

```csharp
using System;
using Xunit;

public class TestingFixture { }

[CollectionDefinition(nameof(DatabaseTestsCollectionFixture))]
public class DatabaseTestsCollectionFixture : ICollectionFixture<TestingFixture>
{ }

[Collection(nameof(DatabaseTestsCollectionFixture))]
public class TestBase : IDisposable
{
	public TestBase(TestingFixture fixture)
	{
		Fixture = fixture;
	}

	public TestingFixture Fixture { get; }

	[Fact]
	public void TestMethod() { }

	public void Dispose()
	{
		GC.SuppressFinalize(this);
	}
}
```

I do not see any analyzer reports. Perhaps you can provide a full repro?

**Comment by bradwilson** [comment]
@rwb196884 It's likely a bug, that's "what the fuck". And please calm down and use appropriate language or I'll end up having to block you from the project. Thanks.

**Comment by bradwilson** [comment]
Also, "through" is spelled with an H.

**Comment by bradwilson** [comment]
Available in `1.8.0-pre.6` https://xunit.net/docs/using-ci-builds

@philippedurocher If you're still having problems that aren't resolved by this build, please open a new issue with a repro project. Thanks!

**Comment by philippedurocher** [comment]
I tested the CI builds and it fixes the error. Thank you! Have a nice day.
